pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedAssignmentRule.java

```diff
@@ -175,6 +175,9 @@ private void reportFinished(GlobalAlgoState result, RuleContext ruleCtx) {
                 if (isIgnorablePrefixIncrement(entry.rhs)) {
                     continue;
                 }
+                if (isUsedLocalVarWithoutInitializer(entry, result.usedVariables)) {
```

Comment:
I don't think we need to maintain a separate set. The problem here is that the "assignment" that is killed for this variable is not really an assignment. If we just don't `assign` the variable with the non-existent value, it will not be reported. I pushed a fix.